From 92edca4fe509a38a9f1e2749ba22f89025f49d61 Mon Sep 17 00:00:00 2001
From: Chaitanya Kulkarni <chaitanya.kulkarni@wdc.com>
Date: Thu, 6 Aug 2020 22:57:34 -0700
Subject: [PATCH 1/7] nvmet: for pt I/O commands use likely for ns check

I/O commands (nvme_cmd_read, nvme_cmd_write) are most common commands
when accessing passthru controller. For I/O commands ns is always
present and the condition is marked as likely. Annotate post request
submission ns check with likely which is dependent on the ns check likely
condition.

Signed-off-by: Chaitanya Kulkarni <chaitanya.kulkarni@wdc.com>
---
 drivers/nvme/target/passthru.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/nvme/target/passthru.c b/drivers/nvme/target/passthru.c
index 8bd7f656e240..18e96eda39b1 100644
--- a/drivers/nvme/target/passthru.c
+++ b/drivers/nvme/target/passthru.c
@@ -268,7 +268,7 @@ static void nvmet_passthru_execute_cmd(struct nvmet_req *req)
 				      nvmet_passthru_req_done);
 	}
 
-	if (ns)
+	if (likely(ns))
 		nvme_put_ns(ns);
 
 	return;
-- 
2.22.1

